spring-pulsar/src/main/java/org/springframework/pulsar/listener/reactive/DefaultReactivePulsarMessageListenerContainer.java

```diff
@@ -179,8 +180,13 @@ private ReactiveMessagePipeline startPipeline(ReactivePulsarContainerProperties<
 					.messageHandler(((ReactivePulsarOneByOneMessageHandler<T>) messageHandler)::received)
 					.handlingTimeout(containerProperties.getHandlingTimeout());
 			if (containerProperties.getConcurrency() > 0) {
-				pipeline = messagePipelineBuilder.concurrent().concurrency(containerProperties.getConcurrency())
-						.maxInflight(containerProperties.getMaxInFlight()).build();
+				ConcurrentOneByOneMessagePipelineBuilder<T> concurrentPipelineBuilder = messagePipelineBuilder
+						.concurrent().concurrency(containerProperties.getConcurrency())
+						.maxInflight(containerProperties.getMaxInFlight());
+				if (containerProperties.isUseKeyOrderedProcessing()) {
+					concurrentPipelineBuilder.useKeyOrderedProcessing();
```

Comment:
**[NOT-DIRECTLY-RELATED-WISHLIST]** The fluent builder syntax is great.... until you have to do something conditionally. I want to explore a builder of the form `field(T)` and `fieldIf(Supplier<T>, Condition<T>)` sometime. 

Ok, please resolve this comment once digested. Just sharing thoughts.

Reply:
cc @lhotari